src/udev/net/ethtool-util.h

```diff
@@ -66,10 +66,21 @@ typedef enum NetDevPort {
         _NET_DEV_PORT_INVALID = -1
 } NetDevPort;
 
+typedef enum NetDevXcvr {
+        NET_DEV_XCVR_INTERNAL = XCVR_INTERNAL,
+        NET_DEV_XCVR_EXTERNAL = XCVR_EXTERNAL,
+        _NET_DEV_XCVR_MAX,
+         _NET_DEV_XCVR_INVALID = -1
+} NetDevXcvr;
+
 #define ETHTOOL_LINK_MODE_MASK_MAX_KERNEL_NU32    (SCHAR_MAX)
 
 /* layout of the struct passed from/to userland */
 struct ethtool_link_usettings {
+        struct {
+                __u8 transceiver;
+        } deprecated;
+
```

Comment:
hmm, is this really right? you are adding one byte at the top of this structure? i mean, this structure is supposed to be the kernel interface, so how can this work? Either this was broken before, or is now, but this is very strange? Can you elaborate?